src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -31,22 +36,36 @@ public void init() {
   }
 
   @Override
-  // Get a specific user. Return null if not found. TODO: User Validation
+  // Get a specific user. Return the currently signed-in user if no userID is
+  // supplied.
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGithub = request.getParameter("githubID");
-    Key userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userGithub);
-    System.out.println(userKey);
 
-    PersonObject userObject;
-    try {
-      userObject = new PersonBuilder().buildPersonObject(datastore.get(userKey));
-      System.out.println(userObject);
-    } catch (Exception e) {
-      e.printStackTrace();
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+    String userID = request.getParameter("userID");
+    if (userID == null) {
+      try {
+        userID =
+            AuthServlet.getAuthorizedUser().getUserId(); // TODO: Should this be a functionality?
+      } catch (NullPointerException e) {
+        ErrorResponse.sendJsonError(
+            response,
+            "UserID not Supplied and user not logged in",
+            HttpServletResponse.SC_FORBIDDEN,
+            "You are not logged in");
+        return; // TODO: Establish Redirect page path
+      }
```

Comment:
nit: Consider using an if statement like
```java
User user = AuthServlet.getAuthorizedUser();
if (user == null) {
  [...]
}
```
instead of the try catch block. Not only do I think it's more readable that way, but it seems that the unspoken rule for try-catches is [this](https://softwareengineering.stackexchange.com/questions/141522/if-else-statements-or-exceptions):

> Typically, I tend to use exceptions for exceptional cases - the instance where nothing should go wrong, and if it does go wrong, that's a big problem.

Where a user not being authorized is a rather common occurrence.

Reply:
Done! It made stuff a lot easier to read.